src/pywifes/splice.py

```diff
@@ -0,0 +1,360 @@
+        ) * fluxVar[overlap]
+
+        flux[red_only] = flux_R[red_only]
+        fluxVar[red_only] = fluxvar_R[red_only]
+
+        return flux, fluxVar
+
+
+def splice_spectra(blue_spec_path, red_spec_path, output_path):
+    """
+    The main routine
+    """
+    # Create to instances of SingleSpec
+    blueSpec = SingleSpec(blue_spec_path)
+    redSpec = SingleSpec(red_spec_path)
+
+    # Join the spectra
+    print("Splicing spectra.")
+    flux, fluxVar = join_spectra(blueSpec, redSpec)
+
+    if flux is None:
+        print("No spectral overlap.")
+
+    else:
+        # Write out the results
+        # Use the header in red arm to start with
+        # Add additional blue CCD keywords as required
+        hdulist = fits.HDUList(fits.PrimaryHDU())
+        hdulist[0].data = flux
+        hdulist[0].header = redSpec.header.copy()
+        hdulist[0].header["CRPIX1"] = blueSpec.header["CRPIX1"]
+        hdulist[0].header["CRVAL1"] = blueSpec.header["CRVAL1"]
+        hdulist[0].header["CDELT1"] = blueSpec.header["CDELT1"]
+        hdulist[0].header["CTYPE1"] = "WAVE"
+        hdulist[0].header["CUNIT1"] = "Angstrom"
+
+        hdr_fluxvar = fits.Header()
+        hdr_fluxvar["EXTNAME"] = "VARIANCE"
+        hdr_fluxvar["CRPIX1"] = blueSpec.header["CRPIX1"]
+        hdr_fluxvar["CRVAL1"] = blueSpec.header["CRVAL1"]
+        hdr_fluxvar["CDELT1"] = blueSpec.header["CDELT1"]
+        hdr_fluxvar["CTYPE1"] = "WAVE"
+        hdr_fluxvar["CUNIT1"] = "Angstrom"
+        hdr_fluxvar["BUNIT"] = "(count / Angstrom)^2"
+
+        hdu_fluxvar = fits.ImageHDU(data=fluxVar, header=hdr_fluxvar)
+        hdulist.append(hdu_fluxvar)
+
+        print("Saving spliced spectra.")
+        hdulist.writeto(output_path, overwrite=True)
+        hdulist.close()
+
+
+def join_cubes(blue_path, red_path):
+
+    # Read red data and metadata
+    red_flux_cube, blue_header = fits.getdata(red_path, 0, header=True)
+    red_fluxvar_cube = fits.getdata(red_path, 1, header=False)
+    red_CRVAL1 = blue_header["CRVAL3"]
+    red_CDELT1 = blue_header["CDELT3"]
+    red_CRPIX1 = blue_header["CRPIX3"]
+    red_NAXIS1 = blue_header["NAXIS3"]
+    red_wavelength = (np.arange(red_NAXIS1) - red_CRPIX1 + 1) * red_CDELT1 + red_CRVAL1
+    red_min_wavelength = min(red_wavelength)
+
+    # Read blue data and metadata
+    blue_flux_cube, red_header = fits.getdata(blue_path, 0, header=True)
+    blue_fluxvar_cube = fits.getdata(blue_path, 1, header=False)
+    blue_CRVAL1 = red_header["CRVAL3"]
+    blue_CDELT1 = red_header["CDELT3"]
+    blue_CRPIX1 = red_header["CRPIX3"]
+    blue_NAXIS1 = red_header["NAXIS3"]
+    blue_wavelength = (
+        np.arange(blue_NAXIS1) - blue_CRPIX1 + 1
+    ) * blue_CDELT1 + blue_CRVAL1
+    blue_max_wavelength = max(blue_wavelength)
+
+    # No wavelength overlap
+    if red_min_wavelength > blue_max_wavelength:
+        return None, None
+
+    # Cubes overlap in wavelength
+    else:
```

Comment:
If a previous `if` results in a function returning/exiting/etc, then you don't *strictly* need an else, since the condition *must* be false if you've reached that code.

That said, it can be hard to keep track of how many conditions are effectively false if there isn't the corresponding code indentation, so opinions vary as to which one is more readable.

In this case, the code below is a bit indented already, and it's all group up pretty neatly, so I'd lean towards removing the 'else'. Either's fine though.